### PR TITLE
Use tsup for building project

### PR DESCRIPTION
**Pull request by fedemp**
Use `tsup` for building the project, thus avoiding all the bash rename magic.

### CODE REVIEW

**Comment by fedemp** [comment]
Hello, @ViktorQvarfordt. Since you mentioned in my previous PR that you were not very fond of the `find` and `mv` magic, I tried using [tsup](https://github.com/egoist/tsup) to build your project and it worked flawlessly.

Only difference is that `dist` files are not split like before.

File: package.json [hunk 1]
```diff
@@ -44,7 +44,9 @@
   "devDependencies": {
     "@types/node": "18.0.3",
     "@types/react": "18.0.15",
+    "react": "latest",
```
Comment (fedemp) [inline package.json:47]: `remix-run` complained about `react` being missing, and it was right. If it's in `peerDependencies`, it must be a dev dependency.

File: tsconfig.json [hunk 2]
```diff
@@ -25,7 +25,6 @@
     "downlevelIteration": true,
     "isolatedModules": true,
     "skipLibCheck": true,
-    "incremental": true,
```
Comment (fedemp) [inline tsconfig.json:28]: `tsup` didn't like this line.

**Comment by ViktorQvarfordt** [comment]
Very cool! Thank you for the improvement. Merging!

**Pull request merged**